### PR TITLE
Separates out downloading dependencies from running tests

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
rebased

File: .travis.yml [hunk 1]
```diff
@@ -21,10 +21,11 @@ cache:
 before_script:
   # default $SBT_OPTS is irrelevant to sbt lancher
   - unset SBT_OPTS
+  # ensure the build errors if dependencies aren't available
+  - travis_retry ./bin/sbt ++$TRAVIS_SCALA_VERSION update
```
Comment (codefromthecrypt) [inline .travis.yml:25]: keyword retry! so whenever maven.twttr.com fails (and yes.. it does often).. we might actually be able to proceed.

Comment (jamescway) [inline .travis.yml:25]: how many times does it retry?  Kind of looks like 3 by default? Looking for reference...

Comment (jamescway) [inline .travis.yml:25]: yeah its [3 times](http://docs.travis-ci.com/user/build-timeouts/#stq=&stp=0)  "Beyond Bundler, you can wrap commands using the function travis_retry which checks the return code of a command, retrying it three times if the return code is non-zero."

**Comment by jamescway** [comment]
:+1:

File: .travis.yml (hunk 1 above)
Comment (codefromthecrypt) [inline .travis.yml:25]: not easy to find, but yeah 3 tries

http://docs.travis-ci.com/user/build-timeouts/

**Pull request merged**